modules/ROOT/pages/_partials/sync/server-auth.adoc

```diff
@@ -1,3 +1,142 @@
+* Integration with GraphQL context - Use the `context` object within the GraphQL resolvers to access user credentials and several helper functions.
+
+= Prerequisites
+
+* There is a Keycloak service available.
+* You must add a valid `keycloak.json` config file to your project. 
+  ** Create a client for your application in the Keycloak administration console.
+  ** Click on the Installation tab.
+  ** Select *Keycloak OIDC JSON* for Format option, and click *Download*.
+
+= Setting Up Keycloak Protection in {sync-server}
+
+1. Import the Voyager Keycloak module
+
+[source,javascript]
+----
+const { KeycloakSecurityService } = require('@aerogear/voyager-keycloak')
+----
+
+2. Read the Keycloak config and pass it to initialise the `KeycloakSecurityService`.
+
+[source,javascript]
+----
+const keycloakConfig = JSON.parse(fs.readFileSync(path.resolve(__dirname, './path/to/keycloak.json')))
+const keycloakService = new KeycloakSecurityService(keycloakConfig)
+----
+
+3. Use the `keycloakService` instance to protect your app:
+
+[source,javascript]
+----
+const app = express()
+keycloakService.applyAuthMiddleware(app)
+----
+
+4. The last piece of work to do is telling Voyager server that the `keycloakService` above will be used as the security service:
+
+[source,javascript]
+----
+const voyagerConfig = {
+  securityService: keycloakService
+}
+const server = VoyagerServer(apolloConfig, voyagerConfig)
+----
+
+The link:https://github.com/aerogear/voyager-server/blob/master/examples/keycloak[Keycloak Example Server Guide] has an example server based off the instructions above and shows all of the steps needed to get it running.
+
+= Using the `@hasRole` Directive in Your Schema
+
+The Voyager Keycloak module provides the `@hasRole` directive to define role based authorisation in your schema. The `@hasRole` directive is a special annotation that can be applied to
+
+* Fields
+* Queries
+* Mutations
+* Subscriptions
+
+The `@hasRole` usage is as follows:
+
+* `@hasRole(role: String)`
+  * Example - `@hasRole(role: "admin"])`
+  * If the authenticated user has the role `admin` they will be authorized.
+* `@hasRole(role: [String])`
+  * Example - `@hasRole(role: ["admin", "editor"])`
+  * If the authenticated user has **at least one of the roles** in the list, they will be authorized.
+
+**The default behaviour is to check client roles.** For example, `@hasRole(role: "admin"])` will check that user has a client role called `admin`. `@hasRole(role: "realm:admin"])` will check if that user has a realm role called `admin` 
+
+The syntax for checking a realm role is `@hasRole(role: "realm:<role>")`. For example, `@hasRole(role: "realm:admin")`. Using a list of roles, it is possible to check for both client and realm roles at the same time.
+
+== Example: Using the @hasRole Directive to Apply Role Based Authorization in a Schema
```

Comment:
not a heading, maybe use following syntax:

.Example: Using ...

to make it pop on the page, without creating toc entry